### PR TITLE
[pipelines] Text2TextGenerationPipeline

### LINKED ISSUE #4411: Pipeline for Conditional Generation (T5 type models)
As text-to-text models (like T5) increase the accessibility of multi-task learning, it also makes sense to have a flexible "Conditional Generation" pipeline.

For example, I should be able to use this pipeline for a multitude of tasks depending on how I format the text input (examples in Appendix D of the [T5 paper](https://arxiv.org/pdf/1910.10683.pdf)). As a baseline, this should be able to work on `T5ForConditionalGeneration` and allow for any of the tasks that are learned by the open sourced T5 model.

Since T5 isn't usable for `TextGenerationPipeline`, I propose we add a `ConditionalGenerationPipeline`.

Please do let me know if there is an existing way to perform the above via pipelines, or if adding a pipeline doesn't makes sense for this; otherwise, I can submit a PR for the above `ConditionalGenerationPipeline` 🙂

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR adds `Text2TextGenerationPipeline`, as discussed here #4411

Fixes #4411

@patrickvonplaten , @LysandreJik 

cc @enzoampil :))

### CODE REVIEW

File: src/transformers/pipelines.py [hunk 1]
```diff
@@ -2049,6 +2051,102 @@ def __call__(
...
+    This Text2TextGenerationPipeline pipeline can currently be loaded from :func:`~transformers.pipeline` using the following
+    task identifier: :obj:`"text2text-generation"`.
+
+    The models that this pipeline can use are models that have been fine-tuned on a translation task.
+    See the up-to-date list of available models on
+    `huggingface.co/models <https://huggingface.co/models?filter=seq2seq>`__.
+
+    Usage::
+        text2text_generator = pipeline("text2text-generation")
+        text2text_generator("question: What is 42 ? context: 42 is the answer to life, the universe and everything")
```
Comment (patil-suraj) [inline src/transformers/pipelines.py:2068]: need good example here, included t5 text2text QA example for now

**Review by LysandreJik — APPROVED** [review]
Very clean implementation, love it!

Could you add the pipeline to the documentation alongside `TextGenerationPipeline`? It can be done in this [file](https://github.com/huggingface/transformers/blob/master/docs/source/main_classes/pipelines.rst).

**Comment by patil-suraj** [comment]
Thanks @LysandreJik ! It was already in the doc at the bottom, now moved it below `TextGenerationPipeline`

**Comment by patil-suraj** [comment]
failing test doesn't look related to this PR

**Review by JetRunner — APPROVED** [review]
Super cool!!

**Review by sgugger — APPROVED** [review]
Thanks a lot, this looks great!

**Review by enzoampil — APPROVED** [review]
Great job on this @patil-suraj ! :smile:

**Review by patrickvonplaten — APPROVED** [review]
This is great @patil-suraj ! Thanks a mille

**Pull request merged**